File: ClientApp/src/app/views/transferPosting/model/emp-transfer-posting.ts
export class EmpTransferPosting {
    id: number = 0;
    empId: number | null = null;

    empIdCardNo: string | null = null;
    empName: string | null = null;
    departmentName: string | null = null;
    designationName: string | null = null;
    sectionName: string | null = null;
    
    approveByIdCardNo: string | null = null;
    approveByEmpName: string | null = null;
    approveByDepartmentName: string | null = null;
    approveByDesignationName: string | null = null;
    approveBySectionName: string | null = null;
    
    deptReleaseByIdCardNo: string | null = null;
    deptReleaseByEmpName: string | null = null;
    deptReleaseByDepartmentName: string | null = null;
    deptReleaseByDesignationName: string | null = null;
    deptReleaseBySectionName: string | null = null;
    deptReleaseByDepartmentId:  number | null  = null;
    deptReleaseBySectionId:  number | null  = null;
    deptReleaseByResponsibilityTypeId:  number | null  = null;
    deptReleaseByDesignationId:  number | null  = null;
    
    joiningReportingByIdCardNo: string | null = null;
    joiningReportingByEmpName: string | null = null;
    joiningReportingByDepartmentName: string | null = null;
    joiningReportingByDesignationName: string | null = null;
    joiningReportingBySectionName: string | null = null;
    joiningReportingByDepartmentId: number | null  = null;
    joiningReportingBySectionId: number | null  = null;
    joiningReportingByResponsibilityTypeId: number | null  = null;
    joiningReportingByDesignationId: number | null  = null;
  

    releaseTypeName: string | null = null;
    deptReleaseTypeName: string | null = null;

    applicationById: number | null = null;
    applicantDepartmentId:  number | null  = null;
  applicantSectionId:  number | null  = null;
  applicantDesignationId:  number | null  = null;
  applicantJobResponsibilityTypeId:  number | null  = null;
    currentOfficeId: number | null = null;
    currentDeptJoinDate: Date | null = null;
    currentDepartmentId: number | null = null;
    currentDesignationId: number | null = null;
    currentSectionId: number | null = null;

    currentGradeId: number | null = null;
    currentScaleId: number | null = null;
    currentBasicPay: number | null = null;
    updateGradeId: number | null = null;
    updateScaleId: number | null = null;
    updateBasicPay: number | null = null;

    withPromotion: boolean = false;
    currentGradeName: string = '';
    currentScaleName: string = '';
    updateGradeName: string = '';
    updateScaleName: string = '';

    officeOrderNo: string | null = null;
    officeOrderDate: Date | null = null;
    orderOfficeById: number | null = null;
    orderByOffice: string = '';

    releaseTypeId: number | null = null;
    transferOfficeId: number | null = null;
    transferDepartmentId: number | null = null;
    transferDesignationId: number | null = null;
    transferDepartmentName: string | null = null;
    transferDesignationName: string | null = null;
    transferSectionName: string | null = null;
    transferSectionId: number | null = null;
    isTransferApprove: boolean = true;

    provideTransferApproveInfo: boolean = false;
    transferApproveById: number | null = null;
    transferApproveDate: Date | null = null;
    approveRemark: string | null = null;
    transferApproveStatus: boolean | null = null;

    isDepartmentApprove: boolean = true;
    provideDepartmentApproveInfo: boolean = false;
    deptReleaseTypeId: number | null = null;
    deptReleaseById: number | null = null;
    deptReleaseDate: Date | null = null;
    referenceNo: string | null = null;
    deptClearance: boolean = true;
    deptRemark: string | null = null;
    deptApproveStatus: boolean | null = null;

    isJoining: boolean = true;
    provideJoiningInfo: boolean = false;
    joiningReportingById: number | null = null;
    joiningDate: Date | null = null;
    joiningRemark: string | null = null;
    joiningStatus: boolean | null = null;

    applicationStatus: boolean | null = null;
    remark: string | null = null;
    menuPosition: number | null = null;
    isActive: boolean | null = null;
    isAdditionalDesignation : boolean | null = null;
    responsibilityTypeId : number | null = null;
    transferResponsibilityTypeId: number | null = null;
    currentResponsibiltyTypeId: number | null = null;
}

File: Hrm.Domain/EmpTransferPosting.cs
using Hrm.Domain.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hrm.Domain
{
    public class EmpTransferPosting : BaseDomainEntity
    {
        public int Id { get; set; }
        public int? EmpId { get; set; }
        public int? ApplicationById { get; set; }
        public int? CurrentOfficeId { get; set; }
        public DateOnly? CurrentDeptJoinDate { get; set; }
        public int? CurrentDepartmentId { get; set; }
        public int? CurrentSectionId { get; set; }
        public int? CurrentDesignationId { get; set; }
        public int? CurrentGradeId { get; set; }
        public int? CurrentScaleId { get; set; }
        public int? CurrentBasicPay { get; set; }
        public string? OfficeOrderNo { get; set; }
        public DateOnly? OfficeOrderDate { get; set; }
        public int? OrderOfficeById { get; set; }
        public int? ReleaseTypeId { get; set; }
        public int? TransferOfficeId { get; set; }
        public int? TransferDepartmentId { get; set; }
        public int? TransferSectionId { get; set; }
        public int? TransferDesignationId { get; set; }
        public int? UpdateGradeId { get; set; }
        public int? UpdateScaleId { get; set; }
        public int? UpdateBasicPay { get; set; }

        public bool? WithPromotion { get; set; }
        public bool? IsTransferApprove { get; set; }
        public int? TransferApproveById { get; set; }
        public DateOnly? TransferApproveDate { get; set; }
        public string? ApproveRemark { get; set; }
        public bool? TransferApproveStatus { get; set; }

        public bool? IsDepartmentApprove { get; set; }
        public int? DeptReleaseTypeId { get; set; }
        public int? DeptReleaseById { get; set; }
        public DateOnly? DeptReleaseDate { get; set; }
        public int? ReferenceNo { get; set; }
        public bool? DeptClearance { get; set; }
        public string? DeptRemark { get; set; }
        public bool? DeptApproveStatus { get; set; }

        public bool? IsJoining { get; set; }
        public int? JoiningReportingById { get; set; }
        public DateOnly? JoiningDate { get; set; }
        public string? JoiningRemark { get; set; }
        public bool? JoiningStatus { get; set; }

        public bool? ApplicationStatus { get; set; }
        public string? Remark { get; set; }
        public int? MenuPosition { get; set; }
        public bool? IsActive { get; set; }

        public virtual EmpBasicInfo? EmpBasicInfo { get; set; }
        public virtual EmpBasicInfo? ApplicationBy { get; set; }
        public virtual EmpBasicInfo? TransferApproveBy { get; set; }
        public virtual EmpBasicInfo? DeptReleaseBy { get; set; }
        public virtual EmpBasicInfo? JoiningReportingBy { get; set; }
        public virtual Department? OrderOfficeBy { get; set; }
        public virtual Office? CurrentOffice { get; set; }
        public virtual Office? TransferOffice { get; set; }
        public virtual Department? CurrentDepartment { get; set; }
        public virtual Department? TransferDepartment { get; set; }
        public virtual Designation? CurrentDesignation { get; set; }
        public virtual Designation? TransferDesignation { get; set; }
        public virtual Section? CurrentSection { get; set; }
        public virtual Section? TransferSection { get; set; }
        public virtual ReleaseType? ReleaseType { get; set; }
        public virtual ReleaseType? DeptReleaseType { get; set; }
        public virtual Grade? CurrentGrade { get; set; }
        public virtual Grade? UpdateGrade { get; set; }
        public virtual Scale? CurrentScale { get; set; }
        public virtual Scale? UpdateScale { get; set; }
    }
}
